app/src/main/java/org/mozilla/focus/session/SessionCallbackProxy.java

```diff
@@ -32,6 +32,7 @@ public void onPageStarted(@NonNull String url) {
         session.setProgress(MINIMUM_PROGRESS);
 
         session.setUrl(url);
+        delegate.onPageStarted(url);
```

Comment:
nit: If you use the Observers, remove this (that's why the delegates weren't attached in the first place).

But your confusion is a good example of why I don't love these observers. :)